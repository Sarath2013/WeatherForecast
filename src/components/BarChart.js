import React from 'react';
import PropTypes from 'prop-types';
import { Chart } from "react-google-charts";

const BarChart = (props) => {
  return (
    <Chart
      height={'240px'}
      chartType="Bar"
      loader={<div className="chart-loader">Loading Chart ...</div>}
      data={(props.temperatureType === 'celsius') ? props.chartData.chartDataC : props.chartData.chartDataF}
      options={{
        legend: { position: 'none' },
      }}
      rootProps={{ 'data-testid': '2' }}
    />
  );
}

BarChart.propTypes = {
  chartData: PropTypes.shape({
    chartDataC: PropTypes.array.isRequired,
    chartDataF: PropTypes.array.isRequired,
    date: PropTypes.string.isRequired,
    displayDate: PropTypes.string.isRequired,
    humidity: PropTypes.number.isRequired,
    tempC: PropTypes.string.isRequired,
    tempF: PropTypes.string.isRequired,
    weather: PropTypes.string.isRequired,
    weatherArr: PropTypes.array.isRequired,
    fragmentsC: PropTypes.array.isRequired,
    fragmentsF: PropTypes.array.isRequired,
  }).isRequired,
  temperatureType: PropTypes.string.isRequired,
}

export default BarChart;