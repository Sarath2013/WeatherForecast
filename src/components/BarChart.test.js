import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import BarChart from './BarChart';

let wrapper, chartData;

describe("Unit Testing - Rendering BarChart with Fahrenheit", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<BarChart chartData={chartData} temperatureType="fahrenheit" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure BarChart details with Fahrenheit', () => {
        expect(wrapper.props().chartType).toEqual('Bar');
        expect(wrapper.props().data).toEqual(chartData.chartDataF);
    });
});

describe("Unit Testing - Rendering BarChart with Celsius", () => {
    beforeAll(() => {
    });

    afterAll(() => {
        wrapper = null
    });

    test('renders snapshot', () => {
        wrapper = shallow(<BarChart chartData={chartData} temperatureType="celsius" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('ensure BarChart details with Celsius', () => {
        expect(wrapper.props().chartType).toEqual('Bar');
        expect(wrapper.props().data).toEqual(chartData.chartDataC);
    });
});

chartData = {
    chartDataC: [["C", "Temp"],
    ["4C", 4],
    ["3C", 3],
    ["1C", 1]],
    chartDataF: [["F", "Temp"],
    ["39F", 39],
    ["37F", 37],
    ["33F", 33]],
    date: "2019-11-12",
    displayDate: "12 Nov 19",
    humidity: 87,
    tempC: "3C",
    tempF: "36F",
    weather: "overcast clouds",
    weatherArr: [{
        count: 2,
        weather: "overcast clouds"
    }, {
        count: 1,
        weather: "light snow"
    }],
    fragmentsC: [3.6100000000000136, 2.25, 0.3900000000000432],
    fragmentsF: [38.498000000000026, 36.05, 32.702000000000076]
}