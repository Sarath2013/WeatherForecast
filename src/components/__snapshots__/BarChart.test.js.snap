// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing - Rendering BarChart with Celsius renders snapshot 1`] = `
<Chart
  chartActions={null}
  chartEvents={null}
  chartType="Bar"
  chartWrapperParams={Object {}}
  className=""
  columns={null}
  controls={null}
  data={
    Array [
      Array [
        "C",
        "Temp",
      ],
      Array [
        "4C",
        4,
      ],
      Array [
        "3C",
        3,
      ],
      Array [
        "1C",
        1,
      ],
    ]
  }
  diffdata={null}
  formatters={null}
  getChartEditor={null}
  getChartWrapper={[Function]}
  graphID={null}
  graph_id={null}
  height="240px"
  legendToggle={false}
  legend_toggle={false}
  loader={
    <div
      className="chart-loader"
    >
      Loading Chart ...
    </div>
  }
  options={
    Object {
      "legend": Object {
        "position": "none",
      },
    }
  }
  render={null}
  rootProps={
    Object {
      "data-testid": "2",
    }
  }
  rows={null}
  spreadSheetQueryParameters={
    Object {
      "gid": 1,
      "headers": 1,
    }
  }
  spreadSheetUrl={null}
  style={Object {}}
  toolbarID={null}
  toolbarItems={null}
/>
`;

exports[`Unit Testing - Rendering BarChart with Fahrenheit renders snapshot 1`] = `
<Chart
  chartActions={null}
  chartEvents={null}
  chartType="Bar"
  chartWrapperParams={Object {}}
  className=""
  columns={null}
  controls={null}
  data={
    Array [
      Array [
        "F",
        "Temp",
      ],
      Array [
        "39F",
        39,
      ],
      Array [
        "37F",
        37,
      ],
      Array [
        "33F",
        33,
      ],
    ]
  }
  diffdata={null}
  formatters={null}
  getChartEditor={null}
  getChartWrapper={[Function]}
  graphID={null}
  graph_id={null}
  height="240px"
  legendToggle={false}
  legend_toggle={false}
  loader={
    <div
      className="chart-loader"
    >
      Loading Chart ...
    </div>
  }
  options={
    Object {
      "legend": Object {
        "position": "none",
      },
    }
  }
  render={null}
  rootProps={
    Object {
      "data-testid": "2",
    }
  }
  rows={null}
  spreadSheetQueryParameters={
    Object {
      "gid": 1,
      "headers": 1,
    }
  }
  spreadSheetUrl={null}
  style={Object {}}
  toolbarID={null}
  toolbarItems={null}
/>
`;
